### ISSUE TITLE
[BUG] Database file remains locked after failed connection

### ISSUE THREAD

**Issue by matix931** [dizitart/nitrite-database#116]
On second bad connection attempt we got exception:
`NO2.2001: database is already opened in other process`

Example code for reproduction:
```File f = new File("some_file.db");
Nitrite dbx = null;
try {
	dbx = Nitrite.builder()
			.filePath(f)
			.openOrCreate("wrong_login", "wrong_password"); //Got exception here
	
} catch(NitriteException ex) {
	//dbx IS NULL here so we can't close connection
}
```

Is there any other way to release file lock?

I think problem is in Security.openSecurely. On the first line of this function you open connection, but if something is wrong with provided credentials you throw an exception leaving connection opened.

**Comment by anidotnet** [comment]
Fix should be available in 3.2.0-SNAPSHOT

**Comment by Bendat** [comment]
> Fix should be available in 3.2.0-SNAPSHOT

This doesn't seem to be working for me with the snapshot.
